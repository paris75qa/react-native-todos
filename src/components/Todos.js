import React from 'react';
import TodoList from "./TodoList";
import TodoFilter from "./TodoFilter";
import TodoInput from "./TodoInput";

const Todos = (props) => {
    return (
      <div>
        <TodoInput onAddTodo={props.onAddTodo} />
        <TodoList todos={props.todos} filter={props.filter} selectItem={props.onTodoClick} />
        <TodoFilter status={props.filter} onSelectFilter={props.filterTodo} />
      </div>
    );
}
export default Todos;
