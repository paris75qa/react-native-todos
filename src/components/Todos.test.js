import React from 'react';
import Todos from './Todos';
import renderer from 'react-test-renderer';

it('Todo renders correctly', () => {
  const tree = renderer.create(
    <Todos onAddTodo={()=>{}} todos={[]} filter='none' onTodoClick={()=>{}} filterTodo={()=>{}}/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
