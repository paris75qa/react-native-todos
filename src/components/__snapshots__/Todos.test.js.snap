// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo renders correctly 1`] = `
<div>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      style={
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "flex": 1,
          "height": 40,
          "paddingLeft": 5,
        }
      }
      value=""
    />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "flex": 0,
            "marginLeft": 5,
            "marginRight": 5,
          },
        ]
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      >
        Ajouter
      </Text>
    </View>
  </View>
  <View />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Filtres: 
    </Text>
    <View
      style={
        Object {
          "flex": 0,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "#CCC",
                "paddingLeft": 10,
                "paddingRight": 10,
              },
              Object {
                "backgroundColor": "#EEE",
              },
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Tous
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "#CCC",
                "paddingLeft": 10,
                "paddingRight": 10,
              },
              "",
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Termin√©
        </Text>
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Array [
              Object {
                "backgroundColor": "#CCC",
                "paddingLeft": 10,
                "paddingRight": 10,
              },
              "",
            ],
          ]
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          A faire
        </Text>
      </View>
    </View>
  </View>
</div>
`;
